dagger-core/src/main/java/io/odpf/dagger/core/source/parquet/splitassigner/ChronologyOrderedSplitAssigner.java

```diff
@@ -16,19 +20,39 @@
 import java.util.concurrent.PriorityBlockingQueue;
 import java.util.stream.Collectors;
 
+import static com.google.api.client.util.Preconditions.checkArgument;
+import static io.odpf.dagger.core.metrics.aspects.ChronologyOrderedSplitAssignerAspects.SPLITS_AWAITING_ASSIGNMENT;
+import static io.odpf.dagger.core.metrics.aspects.ChronologyOrderedSplitAssignerAspects.TOTAL_SPLITS_DISCOVERED;
+import static io.odpf.dagger.core.metrics.aspects.ChronologyOrderedSplitAssignerAspects.TOTAL_SPLITS_RECORDED;
+import static io.odpf.dagger.core.metrics.reporters.statsd.tags.ComponentTags.getSplitAssignerTags;
+
 public class ChronologyOrderedSplitAssigner implements FileSplitAssigner {
     private final PriorityBlockingQueue<InstantEnrichedSplit> unassignedSplits;
     private static final int INITIAL_DEFAULT_CAPACITY = 11;
     private PathParser pathParser;
     private TimeRangePool timeRangePool;
+    private DaggerGaugeManager daggerGaugeManager;
+    private final StatsDErrorReporter statsDErrorReporter;
 
-    private ChronologyOrderedSplitAssigner(Collection<FileSourceSplit> fileSourceSplits, PathParser pathParser, TimeRangePool timeRangePool) {
+    private ChronologyOrderedSplitAssigner(Collection<FileSourceSplit> fileSourceSplits, PathParser pathParser,
+                                           TimeRangePool timeRangePool, SerializedStatsDReporterSupplier statsDReporterSupplier) {
+        this.registerTagsWithMeasurementManagers(statsDReporterSupplier);
```

Comment:
this method is weird, because it's initialising daggerGaugeManager inside. 

we can just inline the code here, we don't need this method.

Reply:
I moved this out because the constructor was getting biggish with function calls and whatnot. Generally I like to keep constructors small for code readability and leave it to the compiler for [inlining whatever it can](https://stackoverflow.com/a/3925068).

Should I try to keep the initialization parts in the constructor and move the function calls to a separate method ?

Reply:
hm. i think we can separate daggerGaugeManager.markValue() calls from the constructor. 
```
    private ChronologyOrderedSplitAssigner(Collection<FileSourceSplit> fileSourceSplits, PathParser pathParser,
                                           TimeRangePool timeRangePool, SerializedStatsDReporterSupplier statsDReporterSupplier) {
        this.pathParser = pathParser;
        this.timeRangePool = timeRangePool;
        this.statsDErrorReporter = new StatsDErrorReporter(statsDReporterSupplier);
        this.unassignedSplits = new PriorityBlockingQueue<>(INITIAL_DEFAULT_CAPACITY, getFileSourceSplitComparator());
        this.daggerGaugeManager = new DaggerGaugeManager(statsDReporterSupplier);
        initAndValidate(fileSourceSplits);
    }


    private void initAndValidate(Collection<FileSourceSplit> fileSourceSplits) {
        StatsDTag[] splitAssignerTags = getSplitAssignerTags();
        daggerGaugeManager.register(splitAssignerTags);
        daggerGaugeManager.markValue(TOTAL_SPLITS_DISCOVERED, fileSourceSplits.size());
        for (FileSourceSplit split : fileSourceSplits) {
            validateAndAddSplits(split);
        }
        daggerGaugeManager.markValue(TOTAL_SPLITS_RECORDED, unassignedSplits.size());
        daggerGaugeManager.markValue(SPLITS_AWAITING_ASSIGNMENT, unassignedSplits.size());
    }
```

Reply:
Will try to implement it 👍

Reply:
Fixed via commit https://github.com/odpf/dagger/pull/174/commits/87ed606524e937e3183d6a9c0ba445eb19d610e2